### PR TITLE
test: Wait for LVM2 RAID to be fully synchronized before pixel test

### LINKED ISSUE #21196: TestStorageLvm2.testRaid pixel failure
The job `fedora-40/updates-testing` failed on commit 6d478ad51585d9d7f8c4567c155bd641635f284f.

Log: https://cockpit-logs.us-east-1.linodeobjects.com/pull-0-6d478ad5-20241031-013118-fedora-40-updates-testing/log.html

### PR DESCRIPTION
**Pull request by mvollmer**
Fixes #21196

### CODE REVIEW

File: test/verify/check-storage-lvm2 [hunk 1]
```diff
@@ -450,6 +450,8 @@ class TestStorageLvm2(storagelib.StorageCase):
         b.wait_js_func("ph_count_check", card + f" .storage-stripe-pv-box-dev:contains('{bn(disk3)}')", 1)
         b.wait_js_func("ph_count_check", card + f" .storage-stripe-pv-box-dev:contains('{bn(disk4)}')", 1)
         b.wait_js_func("ph_count_check", card + f" .storage-stripe-pv-box-dev:contains('{bn(disk1)}')", 1)
+
+        b.wait(lambda: float(self.machine.execute("lvs vgroup0/lvol0 -o sync_percent --noheading")) == 100)
```
Comment (martinpitt) [inline test/verify/check-storage-lvm2:454]: Shouldn't this wait for the rendered percentage display instead? There may still be some lag between what the OS thinks, and what bubbled through udisks/D-Bus/bridge/channel/React?

**Comment by mvollmer** [comment]
Right: https://cockpit-logs.us-east-1.linodeobjects.com/pull-21333-52fc0177-20241126-112937-arch-storage/log.html#102

The text never goes away... hmm.

**Comment by mvollmer** [comment]
Ouch, the volume group needs to be polled when it is actively synching, but the code to do that is in the React component for the volume group itself. When that component is not on the screen, no polling happens. I'll move it to the client.

**Comment by martinpitt** [comment]
@mvollmer sorry, pixel test conflict -- but I really wanted to land that entangled web of f40 refresh plus pixel updates ASAP.

**Comment by martinpitt** [comment]
Otherwise LGTM, thanks for sorting this out!

**Comment by mvollmer** [comment]
> I'll move it to the client.

But we should suspend polling when the Storage page is not visible.

**Review by martinpitt — DISMISSED** [review]
Thanks for fixing this! One question, but that could also be a follow-up.

File: pkg/storaged/client.js [hunk 2]
```diff
@@ -835,11 +835,34 @@ function update_indices() {
     }
 }
 
+let lvm2_poll_timer = null;
+
+function update_lvm2_polling() {
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply pkg/storaged/client.js:840]: Yes.... I was considering this, but then didn't want to drag this our further... but it's the right thing to do so let's do it.

File: test/verify/check-storage-lvm2 [hunk 3]
```diff
@@ -237,7 +237,13 @@ class TestStorageLvm2(storagelib.StorageCase):
         b.wait_text(self.card_row_col("Thinly provisioned LVM2 logical volumes", 1, 3), mount_point_thin)
 
         # prevent unstable usage numbers from failing the pixel test
-        b.assert_pixels('body', "pool-page", mock={".usage-text": "1 / 100 MB"},
+        b.assert_pixels('body', "pool-page",
+                        mock={
+                            ".usage-text": "1 / 100 MB",
+                            self.card_desc("Pool for thinly provisioned LVM2 logical volumes", "Data used"): "-",
+                            self.card_desc("Pool for thinly provisioned LVM2 logical volumes", "Metadata used"): "-",
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply test/verify/check-storage-lvm2:244]: Hmm, so that we catch layout problems because of realistic strings when they happen? Yes, good idea.

**Comment by mvollmer** [comment]
Bah, mocking runs into races. Let's witch to "ignore"!

File: pkg/storaged/client.js [hunk 4]
```diff
@@ -835,11 +835,36 @@ function update_indices() {
...
+            if (vg.NeedsPolling) {
+                vg.Poll();
+            }
+        }
+    }
+
+    if (need_polling && lvm2_poll_timer == null) {
+        lvm2_poll_timer = window.setInterval(poll, 2000);
+        if (for_visibility)
+            poll();
```
Comment (cockpituous) [inline pkg/storaged/client.js:855]: This added line is not executed by any test.

**Pull request merged**